app/controllers/ems_common.rb

```diff
@@ -558,6 +559,37 @@ def recheck_authentication(id = nil)
     @record.authentication_check_types_queue(@record.authentication_for_summary.pluck(:authtype), :save => true)
   end
 
+  def check_compliance(model)
+    # assert_privileges("#{model.name.underscore}_check_compliance")
+    showlist = @lastaction == "show_list"
+    ids = showlist ? find_checked_items : find_current_item(model)
+
+    if ids.empty?
+      add_flash(_("No %{model} were selected for %{task}") % {:model => ui_lookup(:models => model),
+                                                              :task  => "Compliance Check"}, :error)
+    else
+      process_check_compliance(model, ids)
+    end
+
+    showlist ? show_list : show
+    ids.count
+  end
+
+  def process_check_compliance(model, ids)
+    model.where(:id => ids).order("lower(name)").each do |entity|
+      begin
+        entity.check_compliance
+      rescue StandardError => bang
+        add_flash(_("%{model} \"%{name}\": Error during 'Check Compliance': ") %
+                    {:model => ui_lookup(:model => model),
+                     :name  => entity.name} << bang.message,
+                  :error) # Push msg and error flag
+      else
+        add_flash(_("\"%{record}\": Compliance check successfully initiated") % {:record => entity.name})
+      end
+    end
+  end
```

Comment:
These 2 methods are exact duplication from ContainersCommonMixin.
I don't care if in this PR or separately, but let's figure out if there is a common place to put them, or if we even this much logic for simple find-what's-selected-and-run-a-method.

ApplicationController is maybe too broad.
Hosts have a very appealing 2-line implementation:
https://github.com/ManageIQ/manageiq/blob/0b4ec090b/app/controllers/application_controller/ci_processing.rb#L2490-L2493
where [`host_button_operation` itself](https://github.com/ManageIQ/manageiq/blob/0b4ec090b/app/controllers/application_controller/ci_processing.rb#L2532-L2573) does a lot of stuff, presumably host-specific (except it's mixed into ApplicationController :disappointed:).

OK, it makes more sense to have big per-controller operate-on-selected-item(s) method  + small per-operation methods than try to share big per-operation methods across all controllers.
=> IMO it's not worth sharing these methods as-is with ContainersCommonMixin.
=> Let's see together (for separate PR) if we can share logic across operations — check compliance vs scan images — in ContainersCommonMixin.
=> EmsCommon doesn't seem to have operation-agnostic logic.  There is `process_emss(emss, task)` but it's callers do a lot more and `process_emss` itself then branches on `task` :-(.

I say let's merge this.